server/routes/api/user.ts

```diff
@@ -35,16 +36,25 @@ let postApplicationBranchErrorHandler: express.ErrorRequestHandler = (err, reque
 
 let applicationTimeRestriction: express.RequestHandler = async (request, response, next) => {
 	let requestType: ApplicationType = request.url.match(/\/application\//) ? ApplicationType.Application : ApplicationType.Confirmation;
-
-	let openDate: moment.Moment;
-	let closeDate: moment.Moment;
-	if (requestType === ApplicationType.Application) {
-		openDate = moment(await getSetting<Date>("applicationOpen"));
-		closeDate = moment(await getSetting<Date>("applicationClose"));
+	let branchName = request.params.branch as string; // TODO this might not match
+	let branch = (await Branches.BranchConfig.loadAllBranches()).find(b => b.name.toLowerCase() === branchName.toLowerCase()) as (Branches.ApplicationBranch | Branches.ConfirmationBranch);
+	if (!branch) {
+		response.status(400).json({
+			"error": "Invalid application branch"
+		});
+		return;
 	}
-	else {
-		openDate = moment(await getSetting<Date>("confirmationOpen"));
-		closeDate = moment(await getSetting<Date>("confirmationClose"));
+
+	let user = request.user as IUserMongoose;
+
+	let openDate = branch.open;
+	let closeDate = branch.close;
+	if (requestType === ApplicationType.Confirmation && user.confirmationDeadlines) {
```

Comment:
Why is this logic specific to confirmationDeadlines? Why not application deadlines also?

Reply:
this specifically checks for any rolling deadlines that were written to the user model (`user.confirmationDeadlines`). There is no similar functionality for application deadlines